tutorial_pl/README.md

```diff
@@ -0,0 +1,152 @@
+
+Sip potrafi automatycznie generować testy za pomocą pliku Sipfile. Przykładowy plik Sipfile:
+```
+# czas (sekund), po którym przerywa wzorcówkę podczas generowania testów:
+default_time_limit: 15
+static: [
+	# Tu umieść testy zrobione ręcznie (testy przykładowe)
+	# Syntax: <test-range>
+	tag0a
+]
+gen: [
+	# Tu umieść zasady do generowania testów
+	# Syntax: <test-range> <generator> [generator arguments]
+	tag1a-f utils/generatorka 5 10
+	tag2a utils/generatorka 100 100000
+	tag2b utils/generatorka 1000 10000
+	tag3-5a-d utils/generatorka 100000 1000000000
+]
+```
+
+Po zapisaniu pliku Sipfile, komenda `sip gen` wygeneruje wszystkie pliki *.in za pomocą generatorki, a potem wszystkie pliki *.out za pomocą wzorcówki.
+
+# Plik konfiguracyjny Simfile
+
+Wszelkie ustawienia dotyczące paczki znajdują się w pliku Simfile.
+
+Przykładowy, pełny plik Simfile:
+```
+name: Nazwa Zadania
+label: tag
+statement: doc/tag.pdf
+solutions: [prog/tag.cpp, prog/tags1.cpp, prog/tagb1.cpp]
+memory_limit: 256
+checker: check/checker.cpp
+limits: [
+	0 0.4
+
+	1a 0.4
+	1b 0.4
+	
+	2 2.0
+]
+scoring: [
+	0 0
+	1 25
+	2 75
+]
+tests_files: [
+	0 in/tag0.in out/0.out
+	1a in/tag1a.in out/tag1a.out
+	1b in/tag1b.in out/tag1b.out
+	2 in/tagb.in out/tagb.out
+]
+```
+* name: nazwa zadania
+* label: tag zadania
+* memory_limit: limit pamięci (w MB)
+* solutions: lista programów (domyślnie nie trzeba pisać tej linijki)
+* checker: ścieżka do checkerki (domyślnie nie trzeba)
+* limits: dla każdego testu (bez prefixu nazwy testu) maksymalny limit czasu w sekundach (domyślnie sam generuje)
+* scoring: dla każdej grupy testów ilość przyznawanych punktów (domyślnie daje po równo dla każdej grupy testów, tak by się sumowało do 100)
+* test_files: dla każdego testu ścieżka do pliku wejściowego/wyjściowego (to jakiś bezsens, nie opłaca się tego ruszać bo domyślnie i tak nie trzeba)
+
+Ponieważ nazwa zadania, label oraz limit pamięci są jedynymi parametrami, które są obowiązkowe, można je uzupełnić komendą (zamiast edytorem tekstu): `sip name [value]`, `sip label [value]`, `sip mem [value]`. Podobnie istnieją komendy: `sip main-sol [sol]`, `sip statement [value]`, `sip save scoring`.
+
+# Checkerka
+
+Jeżeli trzeba tylko porównać identyczność plików wyjściowych (z dokładnością do spacji/białych znaków), nie trzeba pisać checkerki, ponieważ sim sam taką swoją domyślną wrzuci.
+TODO reszta
+
+# Końcowe kroki
+
+By doczytać o komendach sip'a, wpisz `sip`. By wytestować programy (zobaczyć ile punktów dostają), wpisz `sip test prog`. By zakończyć tworzenie paczki (wrzucić na sima), wpisz `sip zip` (automatyczne czyszczenie pozostałości / skompilowanych programów i skompresowanie paczki).
```

Comment:
```suggestion
By doczytać o komendach sip'a, wpisz `sip`. By wytestować programy (zobaczyć ile punktów dostają), wpisz `sip test pattern`. By zakończyć tworzenie paczki (wrzucić na sima), wpisz `sip zip` (automatyczne czyszczenie pozostałości / skompilowanych programów i skompresowanie paczki).
```
Jeśli patterna nie ma to odpalane jest tylko wzorcowe rozw. Jeżeli pattern jest podany, to odpalne są wrzystkie rozwiązania których ścieżki zawierają pattern jako podciąg np. `sip test tagb.cpp` odpali wszystkie rozwiązania o rozszerzeniu .cpp które wg. konwencji oia są błędne, zaś `sip test 1` odpali wszystkie rozwiązania które mają w nazwie 1.